go/inst/instance_binlog_dao.go

```diff
@@ -700,63 +790,67 @@ func GetNextBinlogCoordinatesToMatch(instance *Instance, instanceCoordinates Bin
 				} else if recordedInstanceRelayLogCoordinates.Equals(&event.Coordinates) {
 					// We've passed the maxScanInstanceCoordinates (applies for relay logs)
 					endOfScan = true
-					log.Debugf("Reached slave relay log coordinates at %+v", recordedInstanceRelayLogCoordinates)
+					log.Debugf("Reached replica relay log coordinates at %+v", recordedInstanceRelayLogCoordinates)
 				} else if recordedInstanceRelayLogCoordinates.SmallerThan(&event.Coordinates) {
-					return nil, 0, log.Errorf("Unexpected problem: relay log scan passed relay log position without hitting it. Ended with: %+v, relay log position: %+v", event.Coordinates, recordedInstanceRelayLogCoordinates)
+					return nil, noMatchedEvents, log.Errorf("Unexpected problem: relay log scan passed relay log position without hitting it. Ended with: %+v, relay log position: %+v", event.Coordinates, recordedInstanceRelayLogCoordinates)
 				}
 				if endOfScan {
 					// end of binary logs for instance:
-					targetMatchCoordinates, err := otherCursor.getNextCoordinates()
+					otherNextCoordinates, err := otherCursor.getNextCoordinates()
 					if err != nil {
-						log.Debugf("Cannot otherCursor.getNextCoordinates(). otherCoordinates=%+v, cached events in cursor: %d; index=%d", otherCoordinates, len(otherCursor.cachedEvents), otherCursor.currentEventIndex)
-						return nil, 0, log.Errore(err)
+						log.Debugf("otherCursor.getNextCoordinates() failed. otherCoordinates=%+v, cached events in cursor: %d; index=%d", otherCoordinates, len(otherCursor.cachedEvents), otherCursor.currentEventIndex)
+						return nil, noMatchedEvents, log.Errore(err)
 					}
+					// Possible good exit point.
 					// No further sanity checks (read the above lengthy explanation)
-					log.Debugf("Reached limit of relay logs for instance, just after %+v. Other coordinates: %+v", lastConsumedEventCoordinates, targetMatchCoordinates)
-					return &targetMatchCoordinates, countMatchedEvents, nil
+					log.Debugf("Reached limit of relay logs for instance, just after %+v. Other coordinates: %+v", lastConsumedEventCoordinates, otherNextCoordinates)
+					return &otherNextCoordinates, countMatchedEvents, nil
 				}
 			}
 
-			instanceEventInfo = event.Info
-			eventCoordinates = event.Coordinates
-			coordinatesStr := fmt.Sprintf("%+v", event.Coordinates)
-			if len(coordinatesStr) > beautifyCoordinatesLength {
-				beautifyCoordinatesLength = len(coordinatesStr)
-			}
-			log.Debugf("> %+v %+v; %+v", rpad(coordinatesStr, beautifyCoordinatesLength), event.EventType, strings.Split(strings.TrimSpace(instanceEventInfo), "\n")[0])
+			instanceEvent = *event // make a physical copy
```

Comment:
You're comparing two events. If you write to one of them and have 2 pointers pointing to the same event both will be the same and that's not what is wanted. Hence taking a physical copy ensures that you compare 2 different events. That was my intention anyway.